src/ModelContextProtocol/Server/McpServer.cs

```diff
@@ -147,11 +147,13 @@ private void SetResourcesHandler(McpServerOptions options)
         }
 
         if (resourcesCapability.ListResourcesHandler is not { } listResourcesHandler ||
-            resourcesCapability.ReadResourceHandler is not { } readResourceHandler)
+            resourcesCapability.ReadResourceHandler is not { } readResourceHandler ||
+            resourcesCapability.ListResourceTemplatesHandler is not { } listResourceTemplatesHandler)
```

Comment:
Trying to discern from the spec... does broadcasting that you support the resources capability imply that resource templates are supported? I believe the answer is yes, in which case this code is correct, just validating.

Reply:
I had just pushed a change here to not require this handler to be set since it was causing issues with more tests.  If we think it should always be implemented, happy to revert to this and fix the other test cases to implement _something_ for this handler where they currently do not.

Reply:
The TypeScript implementation looks to require it:
https://github.com/modelcontextprotocol/typescript-sdk/blob/0fa2397174eba309b54575294d56754c52b13a65/src/server/index.ts#L220-L228

Same for Java:
https://github.com/modelcontextprotocol/java-sdk/blob/a72abf6e33f5e638df6043461511802bfd0e85c4/mcp/src/main/java/io/modelcontextprotocol/server/McpAsyncServer.java#L416-L420

Reply:
Ok let me change things back and fix up the tests that don't specify a handler then.

Reply:
>so it should always be supported (even if returning nothing) if the Resources capability is advertised.

That could mean though that we just have a default handler that returns empty, rather than failing if someone doesn't set it?

Reply:
Ok, I added a new Defaults class inside the McpServerHandlers type with a default implementation for ListResourceTemplatesHandler and fall back to it if there's not one set on the server, but Resource capability is otherwise specified.

Not sure if ya'll like the approach of putting it in a sub-type like this, let me know if you have a specific opinion of how to do it.  I could also just stick the Func impl right inline with the McpServer code instead.

Reply:
> Not sure if ya'll like the approach of putting it in a sub-type like this, let me know if you have a specific opinion of how to do it. I could also just stick the Func impl right inline with the McpServer code instead.

Yeah, I'd suggest just putting it where the default fallback is needed.

Reply:
Ok moved it to be inline instead.